cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +795,83 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_mask_first_n(int64_t context, const char* data, int32_t data_len,
+                                int32_t n_to_mask, int32_t* out_len) {
+  if (data_len <= 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  std::string str(data, data_len);
+  int32_t counter = 0;
+  for (char& c : str) {
```

Comment:
How about implementing this as follows:
* Create an array of size 256 - mapping digits to 'n'; upper case to 'X'; and lower case to 'x'; and the remaining to null
* Take the char value, index into the array
* If you get a null, write input value into output
* else write the value from the array into output

This has only one if condition in the for loop and will be very fast

Reply:
I did a new implementation without std::string, and more simple and efficiente on the for loop